### DISCUSSION TITLE
modcharting problem

### DISCUSSION THREAD

**Discussion by Ibexlol** [KadeArchive/Kade-Engine#1640] [Q&A] [answered]
![Screenshot 2021-08-07 212218](https://user-images.githubusercontent.com/77413092/128616897-ee717bea-dd3f-4580-aa70-ed9ca3235b2f.png)
![Screenshot 2021-08-08 012256](https://user-images.githubusercontent.com/77413092/128616898-6716a870-3bce-441a-afa7-cb60f2098002.png)
im having a slight problem where when i turn off like the swaying on the notes it stays stuck on a certain place
heres the code for it btw

**Comment by ZenusPurity** [comment, marked as answer]
I suggest you add this on your code: 

I have two ways of going about this, 

1. Create a function and put it as your completion function 
function defaultArrows()
        for i = 0, 7 do 
            tweenPosXAngle(i, _G['defaultStrum'..i..'X'], 0, 0.85, i)
            tweenPosYAngle(i, _G['defaultStrum'..i..'Y'], 0, 0.85, i)
        end
end

After that, get the code you have for the swaying notes, let's just use this for example: 
    if swayIntense then 
        for i = 0, 3 do 
            setActorX(_G['defaultStrum'..i..'X'] + 64 * math.sin((currentBeat + i*0)), i)
            setActorY(_G['defaultStrum'..i..'Y'] + 16 * math.cos((currentBeat + i*32) * math.pi),i)
        end 
        for i = 4, 7 do 
            setActorX(_G['defaultStrum'..i..'X'] - 64 * math.sin((currentBeat + i*0)), i)
            setActorY(_G['defaultStrum'..i..'Y'] + 16 * math.cos((currentBeat + i*32) * math.pi),i)
        end 
    end
And add the function with an else statement at the end: 
    if swayIntense then 
        for i = 0, 3 do 
            setActorX(_G['defaultStrum'..i..'X'] + 64 * math.sin((currentBeat + i*0)), i)
            setActorY(_G['defaultStrum'..i..'Y'] + 16 * math.cos((currentBeat + i*32) * math.pi),i)
        end 
        for i = 4, 7 do 
            setActorX(_G['defaultStrum'..i..'X'] - 64 * math.sin((currentBeat + i*0)), i)
            setActorY(_G['defaultStrum'..i..'Y'] + 16 * math.cos((currentBeat + i*32) * math.pi),i)
        end 
    else
       defaultArrows()
    end

Or you could just do what I do 
    if step == 500 then 
       swayIntense = false 
            for i = 0, 7 do 
               tweenPosXAngle(i, _G['defaultStrum'..i..'X'], 0, 0.85, i)
               tweenPosYAngle(i, _G['defaultStrum'..i..'Y'], 0, 0.85, i)
            end
    end

Hope this helps a bit! I recommend the latter solution so you have more control of how fast of slow the notes return

**Reply by Ibexlol** [reply]
thank you so much
